src/libstore/local-store.cc

```diff
@@ -718,7 +718,11 @@ void LocalStore::checkDerivationOutputs(const StorePath & drvPath, const Derivat
                     // somewhat expensive so we do lazily
                     hashesModulo = hashDerivationModulo(*this, drv, true);
                 }
-                StorePath recomputed = makeOutputPath(i.first, hashesModulo->hashes.at(i.first), drvName);
+                auto hashifi = get(hashesModulo->hashes, i.first);
```

Comment:
```suggestion
                auto currentOutputHash = get(hashesModulo->hashes, i.first);
```

(This recurrent `for(auto i : …)` pattern makes things unpleasantly hard to read, let’s not make it worse ;) )